hnix-store-core/tests/NarFormat.hs

```diff
@@ -252,7 +252,9 @@ test_streamManyFilesToNar = HU.testCaseSteps "streamManyFilesToNar" $ \step ->
 
     step "check constant file usage"
     filesPostcount <- countProcessFiles
-    (filesPostcount - filesPrecount) `shouldSatisfy` (< 50)
+    case ((-) <$> filesPostcount <*> filesPrecount) of
+      Nothing -> pure ()
+      Just c -> c `shouldSatisfy` (< 50)
```

Comment:
It is very often encountered.

I tend to use `maybe` for it.
Results in a less hardcoded structure.
Idea is that pattern matching is a hardcode based on the type with a space-structuring coupling.
If no pattern matching, not space structure dependency - no hardcode and layout can be done in any useful way.

Because of that `bool` can be easily promoted into `maybe, & `maybe` to `either` & also their code layout is seamless.

What found - is that patterns in the code structure easier to see. Factoring-out the same structure is also easier.

So far it worked quite well.